src/main/java/seedu/address/model/applicant/Id.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.model.applicant;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an Applicant's Id in the address book.
+ */
+public class Id {
+
+    public static final String MESSAGE_CONSTRAINTS = "Id can take any values, and it should not be blank";
+
+    /*
+     * The first character of the Id must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
+
+    public final String value;
+
+    /**
+     * Constructs an Id object.
+     *
+     * @param id A valid Id string
+     */
+    public Id(String id) {
+        requireNonNull(id);
+        checkArgument(isValidId(id), MESSAGE_CONSTRAINTS);
+        value = id;
+    }
+
+    /**
+     * Returns true if the input string is a valid Id.
+     */
+    public static boolean isValidId(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof seedu.address.model.applicant.Id
+                && value.equals(((seedu.address.model.applicant.Id) other).value));
```

Comment:
If the ID is going to be int (primitive), the comparison has to use '=='. Maybe it would be better to store the int value of ID in the Integer object.